src/constants.ts

```diff
@@ -6,3 +8,6 @@ export const ROUTES: Record<IRouteKey, string> = {
   room: '/room/:id',
   game: '/game/:id'
 };
+
+export const API_ENDPOINT = 'localhost:8080';
+export const socket = socketIO(API_ENDPOINT);
```

Comment:
This instance will be shared amongst all places that need access to sockets. This is needed as there should only ever be 1 instance of the socket client available so that requests and responses all happen on the same connection and to avoid potential memory leaks.